src/test/java/htsjdk/samtools/cram/encoding/ByteArrayLenCodecTest.java

```diff
@@ -0,0 +1,108 @@
+package htsjdk.samtools.cram.encoding;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.cram.io.*;
+import htsjdk.samtools.cram.structure.EncodingID;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ByteArrayLenCodecTest extends HtsjdkTest {
+
+    // this is the only viable way to make a fully-functional ByteArrayLenEncoding
+    // uses a modified version of ByteArrayLenEncoding toByteArray
+
+    private ByteArrayLenEncoding makeEncoding(final int betaIntOffset, final int betaIntBitLimit, final byte valuesExternalBlockId) throws IOException {
+        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+
+            byteArrayOutputStream.write((byte) EncodingID.BETA.ordinal());
+            final byte[] lenBytes = new BetaIntegerEncoding(betaIntOffset, betaIntBitLimit).toByteArray();
+            ITF8.writeUnsignedITF8(lenBytes.length, byteArrayOutputStream);
+            byteArrayOutputStream.write(lenBytes);
+
+            byteArrayOutputStream.write((byte) EncodingID.EXTERNAL.ordinal());
+            final byte[] byteBytes = new byte[]{valuesExternalBlockId};
+            ITF8.writeUnsignedITF8(byteBytes.length, byteArrayOutputStream);
+            byteArrayOutputStream.write(byteBytes);
+
+            ByteArrayLenEncoding bale = new ByteArrayLenEncoding();
+            bale.fromByteArray(byteArrayOutputStream.toByteArray());
+            return bale;
+        }
+    }
```

Comment:
@jmthibault79 It seems like this code is very similar to the actual code in `ByteArrayLenEncoding` (but its slightly different ?) I wonder if we shouldn't refactor `ByteArrayLenEncoding` and `ByteArrayLenCodec` so we can test them directly instead ? Or is that what you're planning to do in the next round ? It almost seems like we should wait for that if its the latter.